stdlib/builtins.pyi

```diff
@@ -1628,7 +1628,14 @@ class UnicodeEncodeError(UnicodeError):
     reason: str
     def __init__(self, __encoding: str, __object: str, __start: int, __end: int, __reason: str) -> None: ...
 
-class UnicodeTranslateError(UnicodeError): ...
+class UnicodeTranslateError(UnicodeError):
+    encoding: str | None
```

Comment:
Is this ever non-`None`? Constructing a new `UnicodeTranslateError` sets it to `None`, and I can't come up with anything that would actually raise `UnicodeTranslateError`.

Reply:
I mean, I quite honestly have no idea. I don't understand why it has the attribute if it's always `None`. But you're right, it's always `None` when it's initialised, and I can't think of *any reason* why someone would change it from `None` to a string after initialising the exception object.

Reply:
Indeed, the `encoding` attribute seems to be more of an accident than something actually useful. CPython doesn't even provide any C functions to access or change it.

```
akuli@akuli-desktop:~/sourcestuff/cpython$ git grep UnicodeTranslateError Python/
Python/codecs.c:    else if (PyObject_TypeCheck(exc, (PyTypeObject *)PyExc_UnicodeTranslateError)) {
Python/codecs.c:        if (PyUnicodeTranslateError_GetEnd(exc, &end))
Python/codecs.c:    else if (PyObject_TypeCheck(exc, (PyTypeObject *)PyExc_UnicodeTranslateError)) {
Python/codecs.c:        if (PyUnicodeTranslateError_GetStart(exc, &start))
Python/codecs.c:        if (PyUnicodeTranslateError_GetEnd(exc, &end))
Python/codecs.c:    else if (PyObject_TypeCheck(exc, (PyTypeObject *)PyExc_UnicodeTranslateError)) {
Python/codecs.c:        if (PyUnicodeTranslateError_GetStart(exc, &start))
Python/codecs.c:        if (PyUnicodeTranslateError_GetEnd(exc, &end))
Python/codecs.c:        if (!(object = PyUnicodeTranslateError_GetObject(exc)))

akuli@akuli-desktop:~/sourcestuff/cpython$ git grep PyUnicodeTranslateError Include/
Include/cpython/pyerrors.h:Py_DEPRECATED(3.3) PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_Create(
Include/cpython/pyerrors.h:PyAPI_FUNC(PyObject *) _PyUnicodeTranslateError_Create(
Include/pyerrors.h:PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_GetObject(PyObject *);
Include/pyerrors.h:PyAPI_FUNC(int) PyUnicodeTranslateError_GetStart(PyObject *, Py_ssize_t *);
Include/pyerrors.h:PyAPI_FUNC(int) PyUnicodeTranslateError_SetStart(PyObject *, Py_ssize_t);
Include/pyerrors.h:PyAPI_FUNC(int) PyUnicodeTranslateError_GetEnd(PyObject *, Py_ssize_t *);
Include/pyerrors.h:PyAPI_FUNC(int) PyUnicodeTranslateError_SetEnd(PyObject *, Py_ssize_t);
Include/pyerrors.h:PyAPI_FUNC(PyObject *) PyUnicodeTranslateError_GetReason(PyObject *);
Include/pyerrors.h:PyAPI_FUNC(int) PyUnicodeTranslateError_SetReason(
akuli@akuli-desktop:~/sourcestuff/cpython$ 
```